lib/state/wallet/logic.dart

```diff
@@ -1793,4 +1792,50 @@ class WalletLogic extends WidgetsBindingObserver {
       launchUrl(Uri.parse(uri), mode: LaunchMode.externalApplication);
     } catch (_) {}
   }
+
+  void requestWalletActions() {
+    _state.walletActionsRequest();
+  }
+
+  Future<void> evaluateWalletActions() async {
+    _state.walletActionsRequest();
+
+    _state.walletActions = [];
+
+    List<ActionButton> actionsToAdd = [];
+
+    actionsToAdd.add(ActionButton(
+      label: 'Vouchers',
+      buttonType: ActionButtonType.vouchers,
+    ));
+
+    try {
+      final isMinter = await _wallet.minter;
+      _state.setWalletMinter(isMinter);
+
+      if (isMinter) {
+        actionsToAdd.add(ActionButton(
+          label: 'Minter',
+          buttonType: ActionButtonType.minter,
+        ));
+      }
+    } catch (_) {}
+
+    List<PluginConfig> plugins = _state.config!.plugins;
```

Comment:
Again, don't read from state. Actually read the config. You can get it from the db, since you have a table for it.